File: backend/app/schemas/rss_assistant.py
from typing import Optional

from pydantic import BaseModel


class ChatMessage(BaseModel):
    role: str
    content: str


class RssRecommendation(BaseModel):
    name: str
    url: str
    description: Optional[str] = None
    suggested_category: Optional[str] = None


class RecommendRequest(BaseModel):
    messages: list[ChatMessage]


class RecommendResponse(BaseModel):
    text: str
    sources: list[RssRecommendation]
    has_config: bool


class BatchCreateRequest(BaseModel):
    sources: list[RssRecommendation]

File: frontend/src/api/rssAssistant.ts
import client from './client'

export interface RssRecommendation {
  name: string
  url: string
  description?: string
  suggested_category?: string
}

export interface ChatMessage {
  role: 'user' | 'assistant'
  content: string
}

export interface RecommendResponse {
  text: string
  sources: RssRecommendation[]
  has_config: boolean
}

export interface BatchCreateRequest {
  sources: RssRecommendation[]
}

export interface BatchCreateResponse {
  created: number
  skipped: number
  results: { name: string; url: string; status: string; reason?: string }[]
}

export function sendChatMessage(messages: ChatMessage[]) {
  return client
    .post<RecommendResponse>('/chat/rss-sources', { messages })
    .then((r) => r.data)
}

export function batchCreateSources(sources: RssRecommendation[]) {
  return client
    .post<BatchCreateResponse>('/sources/batch', { sources })
    .then((r) => r.data)
}
